### PR TITLE
Rewrite non-radix components

### LINKED ISSUE #185: Rewrite non radix components
Capturing the non-radix components as a single chore as the implementation should be straightforward and current functionality should be maintained. 

### Component list

The following list has been identified as not requiring radix (although this may be subject to change):

- [x] ChevronousText
- [x] CodeSnippet
- [x] ControlAlert
- [x] ControlFile
- [x] ControlText
- [x] ControlTextarea
- [x] Copiable
- [x] CopyButton
- [x] Form
- [x] FormSubmit

### Criteria

- TypeScript
- Functional component
- Tests requiring Enzyme should be replaced with react-test-library

### PR DESCRIPTION
**Pull request by tristen**
This PR rewrites components where radix is unnecessary as functional, typescript components and replaces Enzyme with react-testing-library. There are a small number of breaking changes to be aware of:

🚨 **Breaking changes**:

#### CopyButton
- Dropped `textEl` as it was not in use
- Removed `isSupported` check to render the component.
  - Components relying on it like `Copiable` already have this check
  - Consumers can use `CopyButton.isSupported` directly and manage their own fallback behavior

#### ControlFile 
- Dropped `label` as it was not in use
  - Dropped `themeLabel` as a response to dropping `label`
  - Dropped `aside` as a response to dropping `label`
- Rename `initialDisplayValue` with `displayValue`. We are no longer updating the CTA with the file list as implementations had not been doing this.
- Removed the clear implementation. Internal to the component when a file was selected. This should now be maintained by the implementor.

Closes #185

### CODE REVIEW

File: src/components/utils/omit.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ type PropertyValues =
   | { [key: string]: PrimitiveDataTypes };
 
 export default function omit(
-  source: { [key: string]: PropertyValues } = {},
```
Comment (miafan23) [inline src/components/utils/omit.ts:7]: Could you explain a little bit about why this is removed? Just curious about it.

File: src/components/form/validators/validate-date.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import isEmptyControlValue from './is-empty-control-value';
 
-export default function validateDate(value) {
+export default function validateDate(value: any) {
```
Comment (miafan23) [inline src/components/form/validators/validate-date.ts:3]:
```suggestion
export default function validateDate(value: any):string {
```
Maybe add a return type like all other validators?

**Review by miafan23 — CHANGES_REQUESTED** [review]
Amazing work! So many changes! :rocket:
There are a few eslint warnings that I think should be cleared.
image

File: src/components/control-file/control-file.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  /** Pass disabled attribute */
+  disabled: PropTypes.bool,
+  /** If provided, the value of this propery displays as an error message. */
+  validationError: PropTypes.node,
+  /** Overrides the "Select a file" label. */
+  displayValue: PropTypes.string,
+  /** Assembly classes to apply to the button trigger element */
+  themeControlFile: PropTypes.string,
+  /** Assembly classes to apply to the control wrapper */
+  themeControlWrapper: PropTypes.string
```
[the remark this answers is not in the document]
Comment (tristen) [reply src/components/control-file/control-file.tsx:139]: Ah! good catch `themeControlFileClear` is no longer relevant so I've removed it and indicated its now dropped in the changelog.

File: src/components/copy-button/copy-button.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
+import React, { ReactElement, useState, useEffect, HTMLAttributes} from 'react';
```
[the remark this answers is not in the document]
Comment (tristen) [reply src/components/copy-button/copy-button.tsx]: 🤔 I'm not reproducing this in Chrome!

copy

@miafan23 are there any repeatable steps?

[the remark this answers is not in the document]
Comment (tristen) [reply src/components/copy-button/copy-button.tsx]: Ah! I _am_ repeating if the application performs a live reload. If I just reload the page on my own, things work as expected. Is that true for you as well?

[the remark this answers is not in the document]
Comment (miafan23) [reply src/components/copy-button/copy-button.tsx]: It seems not related to the live reload. It does not work after I copied successfully a couple of or even two times.
Reproduce step:
1. Copy first example of copiable
2. Copy second example of copiable 
3. Copy example 1 again and then it does work

https://user-images.githubusercontent.com/9078950/185069780-d3ab1178-cbe7-4a93-a9e0-ca06def08130.mp4

File: src/components/control-file/control-file.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (miafan23) [reply src/components/control-file/control-file.tsx:137]: I also found that these rewritten components are missing the default value in the document. Are they removed intentionally? 
This is original document:
Screen Shot 2022-08-17 at 11 21 39
This is current document:
Screen Shot 2022-08-17 at 11 22 22

[the remark this answers is not in the document]
Comment (tristen) [reply src/components/control-file/control-file.tsx:137]: @miafan23 yeah, that's one of the shortcomings of using react docgen and dropping `static defaultProps` from components. What do you think about just manually writing them into the descriptions instead?

File: src/components/copy-button/copy-button.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (tristen) [reply src/components/copy-button/copy-button.tsx]: @miafan23 Thanks again for this catch. I _think_ I've addressed it by reinitializing the copy button with the same ref element each time

File: src/components/control-file/control-file.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (miafan23) [reply src/components/control-file/control-file.tsx:137]: Yeah, I think it's OK to manually write default values if auto-generating is impossible.

File: src/components/copy-button/copy-button.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (miafan23) [reply src/components/copy-button/copy-button.tsx]: Works now 🥳! Even though I don't not get the reason why it was not working if the clicked button was the ref.

**Review by miafan23 — APPROVED** [review]

File: src/components/copy-button/copy-button.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (tristen) [reply src/components/copy-button/copy-button.tsx]: I might look into testing this a bit better : /

File: src/components/control-file/control-file.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (tristen) [reply src/components/control-file/control-file.tsx:137]: oh! I figured out a way to do this that preserves the default value parsing! I can destructure the props in the component function as well as declare `props` for that `const extraProps = omit(props, propNames);` variable I need.

**Pull request merged**